src/ceph-volume/ceph_volume/systemd/main.py

```diff
@@ -0,0 +1,91 @@
+"""
+This file is used only by systemd units that are passing their instance suffix
+as arguments to this script so that it can parse the suffix into arguments that
+``ceph-volume <sub command>`` can consume
+"""
+import os
+import sys
+import time
+import logging
+from ceph_volume import log, process
+from ceph_volume.exceptions import SuffixParsingError
+
+
+def parse_subcommand(string):
+    subcommand = string.rsplit('-', 1)[-1]
+    if not subcommand:
+        raise SuffixParsingError('subcommand', string)
+    return subcommand
+
+
+def parse_osd_id(string):
+    osd_id = string.split('-', 1)[0]
+    if not osd_id:
+        raise SuffixParsingError('OSD id', string)
+    if osd_id.isdigit():
+        return osd_id
+    raise SuffixParsingError('OSD id', string)
+
+
+def parse_osd_uuid(string):
+    osd_id = '%s-' % parse_osd_id(string)
+    osd_subcommand = '-%s' % parse_subcommand(string)
+    # remove the id first
+    trimmed_suffix = string.split(osd_id)[-1]
+    # now remove the sub command
+    osd_uuid = trimmed_suffix.split(osd_subcommand)[0]
+    if not osd_uuid:
+        raise SuffixParsingError('OSD uuid', string)
+    return osd_uuid
+
+
+def main(args=None):
+    """
+    Main entry point for the ``ceph-volume-systemd`` executable. ``args`` are
+    optional for easier testing of arguments.
+
+    Expected input is similar to::
+
+        ['/path/to/ceph-volume-systemd', '<osd id>-<osd uuid>-<device type>']
```

Comment:
What if the args aren't just unused, but aren't integers are uuids?

Why not just make it @$type-$stuff, and pass that through to 'ceph-volume $type trigger $stuff' (or similar), and drop all of the parsing code in here?  Basically I think just switching $type to the beginning of the '-' separated list instead of the end makes it unnecessary to parse anything but the first -.  The uuid/id parsing would then happen inside the main ceph-volume command with the lvm code which knows what to expect.

This will be harder to change later because all of the existing systemd units in the wild will have the old format, and this script will end up growing weird conditional parsing/logic for all the various $types.  I think with this change it wouldn't have to change at all.